core/trino-main/src/main/java/io/trino/operator/PipelineStats.java

```diff
@@ -447,6 +447,50 @@ public DataSize getPhysicalWrittenDataSize()
         return drivers;
     }
 
+    public PipelineStats pruneIntermediateStats()
```

Comment:
What's the purpose of these `pruneIntermediateXXX` methods?

Reply:
I don't want serialized `TDigest` to be part of query JSON (it's not needed). `OperatorSummaries` appear few times withing query JSON (query level, stage level, pipeline level) and they increase final JSON size (simple scan `189kb` vs `169kb`)

Reply:
That looks like a leaky abstraction. What's an "intermediate structure", generally speaking? How would one know whether a metric needs to implement that behavior? What's the implication of pruning intermediate structures, etc?

Reply:
> What's the implication of pruning intermediate structures, etc?

You cannot merge them anymore. Intermediate is like partial aggregation